frontend/src/pages/PipelineDetails.test.tsx

```diff
@@ -550,6 +550,69 @@ describe('PipelineDetails', () => {
     );
   });
 
+  it(
+    'uses an empty string and does not show error ' +
+      'when pipeline_spec in the response of getPipelineVersion() is undefined',
+    async () => {
+      jest.spyOn(features, 'isFeatureEnabled').mockImplementation(featureKey => {
+        if (featureKey === features.FeatureKey.V2_ALPHA) {
+          return true;
+        }
+        return false;
+      });
+      getV2PipelineVersionSpy.mockResolvedValue({
+        display_name: 'test-pipeline-version',
+        pipeline_id: 'test-pipeline-id',
+        pipeline_version_id: 'test-pipeline-version-id',
+        pipeline_spec: undefined, // empty pipeline_spec
+      });
+      render(<PipelineDetails {...generateProps()} />);
+
+      await waitFor(() => {
+        expect(getV2PipelineVersionSpy).toHaveBeenCalled();
+        // empty template string from empty pipeline_spec and it won't call createGraph()
+        expect(createGraphSpy).toHaveBeenCalledTimes(0);
+      });
+
+      // No errors
+      expect(updateBannerSpy).toHaveBeenCalledTimes(1); // Once to clear banner
```

Comment:
I think an `pipeline_spec = undefined` is an error situation, how do we communicate with users that we have received no pipeline_spec? Silently showing no graph seems a lack of guidance.

Reply:
Add log error for debug